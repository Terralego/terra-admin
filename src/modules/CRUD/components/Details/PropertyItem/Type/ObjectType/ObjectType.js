import React from 'react';
import PropTypes from 'prop-types';
import { getObjectOrderedValue } from '../../../../../services/utils';

import Type from '..';
import NoValue from '../../NoValue';

const ObjectType = ({
  display_value: displayValue,
  schema: { properties },
  ui_schema: uiSchema,
  ui_schema: {
    'ui:order': UIOrder = [],
  } = {},
}) => {
  if (displayValue === null) {
    return <NoValue />;
  }

  const orderedValue = getObjectOrderedValue(displayValue, UIOrder);

  return (
    <div className="details__object">
      {Object.keys(orderedValue).map(value => {
        const { title = value } = properties[value] || {};
        return (
          <div className="details__object-group" key={title}>
            <div className="details__object-key">
              {title} :
            </div>
            <div className="details__object-value">
              <Type
                display_value={orderedValue[value]}
                schema={properties[value]}
                ui_schema={uiSchema[value]}
              />
            </div>
          </div>
        );
      })}
    </div>
  );
};

ObjectType.propTypes = {
  display_value: PropTypes.shape({}),
  schema: PropTypes.shape({
    properties: PropTypes.shape({}),
  }),
  ui_schema: PropTypes.shape({
    'ui:order': PropTypes.array,
  }),
};

ObjectType.defaultProps = {
  display_value: null,
  schema: {
    properties: {},
  },
  ui_schema: {
    'ui:order': [],
  },
};

export default ObjectType;
