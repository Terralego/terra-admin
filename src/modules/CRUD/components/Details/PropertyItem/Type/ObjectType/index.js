import ObjectType from './ObjectType';

export default ObjectType;
