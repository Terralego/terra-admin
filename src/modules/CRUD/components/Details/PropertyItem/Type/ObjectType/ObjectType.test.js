
import React from 'react';
import renderer from 'react-test-renderer';
import ObjectType from './ObjectType';

jest.mock('..', () => props => <div {...props} />);
jest.mock('../../NoValue', () => () => <div>No value</div>);

const props = {
  display_value: {
    date: '2019',
    source: 'Requalification du site',
  },
  schema: {
    properties: {
      date: {
        type: 'string',
        title: 'Date',
        format: 'date',
      },
      source: {
        type: 'string',
        title: 'Source',
      },
    },
    type: 'object',
  },
};

describe('should render correctly', () => {
  it('object', () => {
    const tree = renderer.create((
      <ObjectType {...props} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('nothing', () => {
    const tree = renderer.create((
      <ObjectType {...props} display_value={null} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
