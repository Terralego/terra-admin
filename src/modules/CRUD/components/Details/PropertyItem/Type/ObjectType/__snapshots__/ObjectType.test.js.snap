// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly nothing 1`] = `
<div>
  No value
</div>
`;

exports[`should render correctly object 1`] = `
<div
  className="details__object"
>
  <div
    className="details__object-group"
  >
    <div
      className="details__object-key"
    >
      Date
       :
    </div>
    <div
      className="details__object-value"
    >
      <div
        display_value="2019"
        schema={
          Object {
            "format": "date",
            "title": "Date",
            "type": "string",
          }
        }
      />
    </div>
  </div>
  <div
    className="details__object-group"
  >
    <div
      className="details__object-key"
    >
      Source
       :
    </div>
    <div
      className="details__object-value"
    >
      <div
        display_value="Requalification du site"
        schema={
          Object {
            "title": "Source",
            "type": "string",
          }
        }
      />
    </div>
  </div>
</div>
`;
